nkms_eth/project/contracts/MinersEscrow.sol

```diff
@@ -411,12 +413,13 @@ contract MinersEscrow is Issuer {
     /**
     * @notice Mint tokens for sender for previous periods if he locked his tokens and confirmed activity
     **/
-    function mint() external onlyTokenOwner {
+    function mint() public onlyTokenOwner {
         uint256 previousPeriod = getCurrentPeriod().sub(uint(1));
         MinerInfo storage info = minerInfo[msg.sender];
         uint256 numberPeriodsForMinting = info.confirmedPeriods.length;
-        require(numberPeriodsForMinting > 0 &&
-            info.confirmedPeriods[0].period <= previousPeriod);
+        if (numberPeriodsForMinting == 0 || info.confirmedPeriods[0].period > previousPeriod) {
```

Comment:
`info.confirmedPeriods[0].period > previousPeriod`

huh?

Reply:
This line means that there is no confirmed previous period, only future periods or current